--- keypair.c	2024-04-10 02:13:57.959428172 -0400
+++ keypair_patched.c	2024-04-10 02:17:26.697736315 -0400
@@ -1,15 +1,13 @@
 #include <string.h>
-#include "randombytes.h"
-#include "crypto_sign.h"
-#include "crypto_hash_sha512.h"
+#include "sha512/sha512.h"
 #include "ge.h"
 
-int crypto_sign_keypair(unsigned char *pk,unsigned char *sk)
+int ed25519_keypair(const unsigned char *seed, unsigned char *pk,unsigned char *sk)
 {
   unsigned char az[64];
   ge_p3 A;
 
-  randombytes(sk,32);
+	memmove(sk, seed, 32);
   crypto_hash_sha512(az,sk,32);
   az[0] &= 248;
   az[31] &= 63;
